src/xpi.js

```diff
@@ -102,4 +102,43 @@ export default class Xpi {
         .catch((err) => reject(err));
     });
   }
+
+  getFileAsString(path) {
+    return new Promise((resolve, reject) => {
+      return this.getFileAsStream(path)
+        .then((fileStream) => {
+          var fileString = '';
+          fileStream.on('data', (chunk) => {
+            fileString += chunk;
+          });
+
+          // Once the file is assembled, resolve the promise.
+          fileStream.on('end', () => {
+            resolve(fileString);
+          });
+        })
+        .catch((err) => reject(err));
+    });
+  }
+
+  getJSFiles() {
+    return new Promise((resolve, reject) => {
+      return this.getMetaData()
+        .then((metadata) => {
+          let jsFiles = [];
+
+          for (let filename in metadata) {
+            // TODO: Check for JS files better than this (follow require path)
```

Comment:
Yeah, it's likely something for a separate issue and could be its own can of worms.

A brute-force way would be to get ESLint to try and parse every file, and anything that throws a parsing error is assumed to not be JS. There are edge cases for that implementation as well, but it would prevent people sneaking code past the validator by naming every file `.javascript` or something.